### PR TITLE
contrib: outline purpose of download-frozen-images-v2.sh

### LINKED ISSUE #40857: download-frozen-image-v2.sh (branch 19.03/master) does not download from non-Docker registries
**Description**
When trying to download container images from non-docker registries (k8s.gcr.io, quay.io, registry.centos.org), download-froxen-image-v2.sh returns "curl: (22) The requested URL returned error: 401 Unauthorized"

**Steps to reproduce the issue:**
1. ```wget https://github.com/moby/moby/blob/19.03/contrib/download-frozen-image-v2.sh```
2. ```chmod +x download-frozen-image-v2.sh```
3. ```./download-frozen-image-v2.sh "kube-apiserver" k8s.gcr.io/kube-apiserver:v1.18.2```

**Describe the results you received:**
```curl: (22) The requested URL returned error: 401 Unauthorized```

**Describe the results you expected:**
I expected successful completion of the script with the output looking similar to:
```
Downloading 'k8s.gcr.io/kube-apiserver:v1.18.2' (1 layers)...

######################################################################## 100.0%

Download of images into 'kube-apiserver' complete.
Use something like the following to load the result into a Docker daemon:
  tar -cC 'kube-apiserver' . | docker load
```

**Additional information you deem important (e.g. issue happens only occasionally):**
I am able to successfully pull images with download-frozen-image-v2.sh from Docker Hub

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:27:04 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b
  Built:            Wed Mar 11 01:25:42 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Output of `docker info`:**

```
lient:
 Debug Mode: false

Server:
 Containers: 17
  Running: 17
  Paused: 0
  Stopped: 0
 Images: 9
 Server Version: 19.03.8
 Storage Driver: overlay2
  Backing Filesystem: <unknown>
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: systemd
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 3.10.0-957.27.2.el7.x86_64
 Operating System: CentOS Linux 7 (Core)
 OSType: linux
 Architecture: x86_64
 CPUs: 3
 Total Memory: 7.638GiB
 Name: 1588216178.test
 ID: ZOYK:SR43:OIMO:ZT36:WBGK:ALKP:WDHV:4NYF:QPN5:GLKF:ZFPB:TPFY
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Docker is not installed on the device that I am using download-frozen-image-v2.sh on.

### LINKED ISSUE #43421: Download a big-dash-name docker image with download-frozen-image-v2.sh
397B9272-9A33-4CD0-8A30-7BE14A83EA32

The docker image is almost 12GB.

### PR DESCRIPTION
**Pull request by thaJeztah**
- closes https://github.com/moby/moby/issues/40857
- closes https://github.com/moby/moby/issues/43421
- closes https://github.com/moby/moby/issues/40857
- related: https://github.com/moby/moby/pull/45229

Using @tianon's suggestion as starting point; https://github.com/moby/moby/issues/40857#issuecomment-629414797

We occassionally receive contributions to this script that are outside its intended scope. Let's add a comment to the script that outlines what it's meant for, and a link to a GitHub ticket with alternatives.

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by rumpl — APPROVED** [review]

File: contrib/download-frozen-image-v2.sh [hunk 1]
```diff
@@ -1,6 +1,19 @@
 #!/usr/bin/env bash
 set -eo pipefail
 
+# This script was developed for use in Moby's CI, and as such the use cases and
```
Comment (AkihiroSuda) [inline contrib/download-frozen-image-v2.sh:4]: Nit: this paragraph may appear before set -eo pipefail

**Review by AkihiroSuda — APPROVED** [review]

File: contrib/download-frozen-image-v2.sh (hunk 1 above)
Comment (thaJeztah) [reply contrib/download-frozen-image-v2.sh:4]: oh! TBH, I wasn't 100% sure if that would mess up things, so I took the "safe" approach ☺️

**Comment by crazy-max** [comment]
Maybe we could move the script in `hack` folder to avoid any confusion? Also recall we had https://github.com/moby/moby/pull/44005/commits/1e109657b20a5575294fe53983443e92eeb37a4a to get rid of this script but was not retained.

**Comment by thaJeztah** [comment]
Yeah perhaps we could at some point. That said, I _know_ there's been various external consumers of the script, and that's perfectly fine if it works for them (in its current form), so I don't see a _direct_ need to break them.

**Review by vvoland — APPROVED** [review]

**Pull request merged**